### PR TITLE
Add base directory and pnpm support

### PR DESCRIPTION
**Pull request by apazzolini**
Support for the configuration we need in the new monorepo structure.

### CODE REVIEW

**Comment by apazzolini** [comment]
Note that we'll need to keep our fork alive until your PR is merged as well as this https://github.com/coinbase/salus/issues/445 issue is addressed.

File: lib/salus/scanners/npm_audit.rb
Comment (Piccirello) [inline lib/salus/scanners/npm_audit.rb:17]: This looks like it breaks npm support. Is that desired? I'd expect this to be additive.

Comment (apazzolini) [reply lib/salus/scanners/npm_audit.rb:17]: Yeah, I took a shortcut in the interest of time since we won't need npm support on our private fork. I can make it additive, but will take some time to figure out how to crawl up from the basedir to find the pnpm lockfile (instead of returning true) and all of the necessary places to hook into to provide pnpm as a standalone thing. Happy to do it if you think it's time well spent, but I was hoping the coinbase maintainers would do it in the issue I opened.

Comment (Piccirello) [reply lib/salus/scanners/npm_audit.rb:17]: For the specific purposes of our fork I think this is ok. It would be nice if we could submit a PR to Salus rather than just an issue, but I also understand that we have higher priority things to work on. I'm not super optimistic we'll be able to get off of our fork any time soon though 😢

**Review by Piccirello — APPROVED** [review]

**Pull request merged**
